### PR TITLE
conductor: restart websocket connection

### PR DESCRIPTION
**Pull request by joroshiba**
## Summary
Recreates websocket connection instead of destroying driver when connection is closed.

## Background
Websockets generally have timeouts, we need to seamlessly spin re-attempt to connect.

## Changes
- move creation of sequencer client into a function instead of having be part of Driver startup
- create client when run starts, create new client if the connection closes.

## Testing
Manually tested within dev-cluster.

## Issues
closes astriaorg/astria#451

### CODE REVIEW

**Review by sambukowski — APPROVED** [review]

File: crates/astria-conductor/src/driver.rs [hunk 1]
```diff
@@ -168,8 +162,14 @@ impl Driver {
                     if let Some(block) = new_block {
                         self.handle_new_block(block).await
                     } else {
-                        warn!("sequencer new-block subscription closed unexpectedly; shutting down driver");
-                        break;
+                        warn!("sequencer new-block subscription closed; restarting");
+                        match self.get_new_block_stream().await {
+                            Ok(res) => new_blocks = res,
+                            Err(_) => {
```
Comment (noot) [inline crates/astria-conductor/src/driver.rs:168]: log this error as part of the `warn`

File: crates/astria-conductor/src/driver.rs [hunk 2]
```diff
@@ -168,8 +162,14 @@ impl Driver {
...
+                                warn!("sequencer new-block restart failed; shutting down driver");
```
Comment (noot) [inline crates/astria-conductor/src/driver.rs:169]:
```suggestion
                                warn!("sequencer new-block subscription restart failed; shutting down driver");
```

File: crates/astria-conductor/src/driver.rs [hunk 3]
```diff
@@ -185,6 +185,18 @@ impl Driver {
...
     }
 
+    async fn get_new_block_stream(&self) -> Result<NewBlocksStream> {
+        use sequencer_client::SequencerSubscriptionClientExt as _;
+
+        let sequencer_client = SequencerClient::new(&self.sequencer_ws_url)
+            .await
+            .wrap_err("failed constructing a cometbft websocket client to read off sequencer")?;
+
+        let block_stream = sequencer_client.client.subscribe_new_block_data().await?;
```
Comment (noot) [inline crates/astria-conductor/src/driver.rs:195]: wrap_err

**Review by noot — APPROVED** [review]

**Comment by SuperFluffy** [comment]
This is addressed as part of #455. Note that the present PR does not address a failure of the underlying websocket connection: since `WebSocketClient` is just a thin wrapper of an mpsc sender to communicate with the websocket driver that is managing the communication with the remote host, the entire TCP connection needs to be reestablished because the mpsc channel is dropped after the driver goes down.

**Pull request closed without merging**